src/DynamoCore/Graph/Nodes/DummyNode.cs

```diff
@@ -546,7 +546,7 @@ private XmlElement OriginalXmlNodeContent
                     Console.WriteLine(args.ErrorContext.Error);
                 },
                 ReferenceLoopHandling = ReferenceLoopHandling.Ignore,
-                TypeNameHandling = TypeNameHandling.Auto,
+                TypeNameHandling = TypeNameHandling.None,
```

Comment:
Can this have unintended consequences or is it safe not serializing the type since the dummy node is not expected to run anyway?

Reply:
note that these settings are only used for deserialization, not serialization.

Reply:
So basically this warning right?
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2326

Reply:
So Auto will actually try to instantiate/resolve type that was specified in the serialized data ?
Basically if someone puts in another type (MyMaliciousAssembly.MyMaliciousType) in the serialized data, the Newtosoft deserializer would try to load the MyMaliciousAssembly assembly and instantiate the MyMaliciousType type ?

Reply:
Also @mjkkirschner any foreseeable issues with nodes that were already serialized with type information ?
Will the new deserialization logic (with TypeNameHandling.None) work just as well ?

What if a serialized node had type references to an assembly on disk (in a package folder) that was not preloaded by Dynamo. Would that break now ?

Reply:
`Auto` is a bit more nuanced than that, it will allow deserialization of types that are of a different type than the one that you declared, so it allows deserialzing concrete classes of abstract types or inherited classes, like for example all the `NodeModel` types - so it seems like a good fit.

I am not sure it will actually load the assembly if it's not already loaded. The gadgets are dangerous because they rely on base class types like `ObjectDataProvider`.

It's unnecessary because we actually load the assembly and types that are defined in the NodeModel's json using `Type.GetType()` and we require node authors to provide constructors marked with `[JsonConstructor]`.